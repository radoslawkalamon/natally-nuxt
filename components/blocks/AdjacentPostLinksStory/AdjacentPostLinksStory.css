.adjacent-post-links-story {
  display: flex;
  gap: var(--gap);
  justify-content: space-between;
}

.adjacent-post-links-story__previous {
  margin-right: auto;
}

.adjacent-post-links-story__next {
  margin-left: auto;
}
