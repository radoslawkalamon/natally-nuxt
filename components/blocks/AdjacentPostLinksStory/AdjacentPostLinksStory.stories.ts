import type { Meta, Story } from '@storybook/vue'
import BlocksAdjacentPostLinksStory from '@/components/blocks/AdjacentPostLinksStory/AdjacentPostLinksStory.vue'
import { storybookOnlyDevTemplate } from '@/devtools/storybook.onlyDev.template'

const meta: Meta = {
  title: 'Blocks / Adjacent Post Links Story',
  component: BlocksAdjacentPostLinksStory
}
export default meta

const Template: Story = (_args, { argTypes }) => ({
  components: {
    BlocksAdjacentPostLinksStory
  },
  props: Object.keys(argTypes),
  template: process.env.NODE_ENV !== 'production'
    ? '<div class="scoped__storybook-section"><BlocksAdjacentPostLinksStory v-bind="$props" /></div>'
    : storybookOnlyDevTemplate
})

export const Default: Story = Template.bind({})
Default.args = {
  slug: 'bajka'
}
