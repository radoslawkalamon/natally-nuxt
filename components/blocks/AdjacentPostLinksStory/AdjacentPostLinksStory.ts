import Vue from 'vue'
import { DTOMetaPostStory, DTOMetaPostStoryConstructor } from '@/utils/dto.meta.post.story'

export default Vue.extend({
  name: 'BlocksAdjacentPostLinksStory',
  components: {
    ComponentsAdjacentPostLinkNext: () => import(/* webpackChunkName: "components-adjacent-post-link-next" */'@/components/components/AdjacentPostLinkNext/AdjacentPostLinkNext.vue'),
    ComponentsAdjacentPostLinkPrevious: () => import(/* webpackChunkName: "components-adjacent-post-link-previous" */'@/components/components/AdjacentPostLinkPrevious/AdjacentPostLinkPrevious.vue'),
    ComponentsAdjacentPostLinkPlaceholder: () => import(/* webpackChunkName: "components-adjacent-post-link-placeholder" */'@/components/components/AdjacentPostLinkPlaceholder/AdjacentPostLinkPlaceholder.vue')
  },
  props: {
    slug: {
      type: String,
      required: true
    }
  },
  data (): Record<string, DTOMetaPostStory | null> {
    return {
      previousStory: null,
      nextStory: null
    }
  },
  async fetch () {
    const articles = await this.$content('opowiadania')
      .sortBy('createdAt', 'asc')
      .surround(this.slug)
      .fetch<DTOMetaPostStoryConstructor>()

    if (Array.isArray(articles)) {
      this.previousStory = articles[0] ? new DTOMetaPostStory(articles[0]) : null
      this.nextStory = articles[1] ? new DTOMetaPostStory(articles[1]) : null
    }
  }
})
