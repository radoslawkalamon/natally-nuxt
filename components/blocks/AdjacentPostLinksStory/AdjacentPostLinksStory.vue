<template>
  <div class="adjacent-post-links-story">
    <ComponentsAdjacentPostLinkPrevious
      v-if="previousStory"
      class="adjacent-post-links-story__previous"
      label="Poprzednie"
      :path="previousStory.path"
      :title="previousStory.title"
    />
    <ComponentsAdjacentPostLinkNext
      v-if="nextStory"
      class="adjacent-post-links-story__next"
      label="NastÄ™pne"
      :path="nextStory.path"
      :title="nextStory.title"
    />
  </div>
</template>

<script src="./AdjacentPostLinksStory.ts" lang="ts"></script>
<style src="./AdjacentPostLinksStory.css" scoped></style>
