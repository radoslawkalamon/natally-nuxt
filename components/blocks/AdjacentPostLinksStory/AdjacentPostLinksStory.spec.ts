import merge from 'lodash/merge'
import { mount } from '@vue/test-utils'
import AdjacentPostLinksStory from './AdjacentPostLinksStory.vue'
import { JestMockNuxtContent } from '@/devtools/jest.mock.nuxt.content'

jest.mock('@/utils/dto.meta.post.story', () => ({
  DTOMetaPostStory: class MockEmptyClass {
    path: string;
    title: string;

    constructor () {
      this.path = '/opowiadania/test-story'
      this.title = 'Test story'
    }
  }
}))

const defaultOptionsFactory = (options?: object) => merge({
  mocks: {
    $content: JestMockNuxtContent(Array(2).fill({}))
  },
  propsData: {
    slug: 'test-slug'
  },
  stubs: [
    'ComponentsAdjacentPostLinkPrevious',
    'ComponentsAdjacentPostLinkNext'
  ]
}, options)

describe('Blocks / Adjacent Post Links Story', () => {
  test('shall render', async (): Promise<void> => {
    const wrapper = mount(AdjacentPostLinksStory, defaultOptionsFactory())
    await (AdjacentPostLinksStory as any).options.fetch.call(wrapper.vm)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
