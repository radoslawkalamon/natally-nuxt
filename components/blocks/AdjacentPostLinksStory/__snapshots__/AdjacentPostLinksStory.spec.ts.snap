// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blocks / Adjacent Post Links Story shall render 1`] = `
"<div class=\\"adjacent-post-links-story\\">
  <componentsadjacentpostlinkprevious-stub label=\\"Poprzednie\\" path=\\"/opowiadania/test-story\\" title=\\"Test story\\" class=\\"adjacent-post-links-story__previous\\"></componentsadjacentpostlinkprevious-stub>
  <componentsadjacentpostlinknext-stub label=\\"NastÄ™pne\\" path=\\"/opowiadania/test-story\\" title=\\"Test story\\" class=\\"adjacent-post-links-story__next\\"></componentsadjacentpostlinknext-stub>
</div>"
`;
